### ISSUE TITLE
Add Job.context

### ISSUE THREAD

**Issue by wslulciuc** [MarquezProject/marquez#632]
Let's add a **context** field defined as a `Map` to [Job](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/service/models/Job.java). A context can be used for getting additional details about a job. The context should be associated with a _job version_ and used in the versioning function applied to job metadata, see [`JobMeta.version()`](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/service/models/JobMeta.java#L58).

### Example

Take the simple case where we collect metadata on a batch job that executes the following SQL:

```
SELECT * FROM room_bookings WHERE room_num = '2H';
```

The SQL value above would be stored in `Job.context` under the key `sql`.

**`METADATA`**

```
{
  "type": "BATCH",
  “name”: “room_bookings_7_days”,
  "inputs": [
    "wedata.room_bookings"
  ],
  "outputs": [],
  "location": "https://github.com/wework/jobs/commit/124f6089ad4c5fcbb1d7b33cbb5d3a9521c5d32c",
  "context": {
    "sql": "SELECT * FROM room_bookings WHERE room_num = '2H';"
  }
  "description": "Determine weekly room booking occupancy patterns."
}
```

### Open questions

1. Should we limited the number of allowed key/value pairs in the context (maybe <= `100`?)

/cc @julienledem @roaraya8 @fanrukun

**Comment by julienledem** [comment]
job context should probably be limited in total size rather that property count. There's going to be underlying storage limitations

**Comment by julienledem** [comment]
this looks good to me.

**Comment by wslulciuc** [comment]
Sure thing, @julienledem! I'll add validation around total size.

**Comment by sshah-wework** [comment]
Neat! Is the plan to store as KV pairs in the DB?

**Comment by wslulciuc** [comment]
@sshah-wework, when I first thought about how we might store the KV pairs representing `Job.context`, I immediately wanted to use `hstore`. But, there are some limitations. You first need to run the following on our PG instance to install the package:

```
CREATE EXTENSION hstore;
```
see, http://postgresguide.com/cool/hstore.html

Then, you need to enable it for [Jdbi]( http://jdbi.org/#_hstore).

I feel we're fine with `JSON`. This also allows the application to be less dependent on the choice of DB. Thoughts?

**Comment by wslulciuc** [comment]
Fixed #636
